server/mqtt.go

```diff
@@ -2676,20 +2676,23 @@ func (sess *mqttSession) clear() error {
 	sess.pubRelConsumer = nil
 	sess.seq = 0
 	sess.tmaxack = 0
-	sess.mu.Unlock()
 
 	for _, dur := range durs {
 		if _, err := sess.jsa.deleteConsumer(mqttStreamName, dur); isErrorOtherThan(err, JSConsumerNotFoundErr) {
+			sess.mu.Unlock()
```

Comment:
This means we are holding the lock across and request/reply boundary, which will be hopping between servers and routes. So do not think we should do that, unless I am mistaken on deleteConsumer. But that is also why we do not hold below for deleteMsg.

Reply:
@derekcollison Understood, that's why I made the change to begin with, in https://github.com/nats-io/nats-server/pull/4349/files#diff-ef6be3a4d0880e5771af5ea89b210064b3f988fdde90076a188b3b20360d8a13R2664 but it was not really related to QoS2; creeped in.

This PR would revert to how it was in v2.9

Anyway, we can hold off on this PR, see if https://github.com/nats-io/nats-server/pull/4734 addresses session `clear()` concerns.